packages/block-library/src/navigation-link/index.php

```diff
@@ -360,12 +382,6 @@ function register_block_core_navigation_link() {
 		}
 	}
 
-	register_block_type_from_metadata(
-		__DIR__ . '/navigation-link',
-		array(
-			'render_callback' => 'render_block_core_navigation_link',
-			'variations'      => array_merge( $built_ins, $variations ),
-		)
-	);
+	$navigation_block_type->variations = array_merge( $navigation_block_type->variations, $built_ins, $variations );
 }
-add_action( 'init', 'register_block_core_navigation_link' );
+add_action( 'init', 'register_block_core_navigation_link_variations', 1000 );
```

Comment:
Speaking away from Github with @ironprogrammer we discussed avoiding using `1000` here. This is because - as you noted - we cannot assume extenders will have done their work by this point.

Another option might be moving this to a later [action hook](https://codex.wordpress.org/Plugin_API/Action_Reference)?  

@ironprogrammer may be able to advise on a suitable candidate but you could experiment in the meantime to see what works 🙏

Reply:
I was thinking, that maybe we could run the initial registration on 10 like it is in core atm but additionally add an action hook to `registered_post_type` and then add the variation in that callback. This way we can be sure, to catch all post types that will be regestered.